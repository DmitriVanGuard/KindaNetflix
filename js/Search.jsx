// @flow

import React from 'react';
import { connect } from 'react-redux';
import ShowCard from './ShowCard';
import Header from './Header';

type Props = {
	shows: Array<Show>,
	searchTerm: string
};

const Search = (props: Props) => (
	<div className="search">
		<Header showSearch />
		{/* Only need now showSearch because with Redux store can directly access searchTerm and handleSearchTermStateChange inside Header component */}
		<div>
			{props.shows
				.filter(show =>
					`${show.title} ${show.description}`.toUpperCase().includes(props.searchTerm.toUpperCase())
				)
				.map(show => (
					<ShowCard key={show.imdbID} {...show /* anti pattern. Better to be explicit */} />
				))}
		</div>
	</div>
);

const mapStateToProps = state => ({ searchTerm: state.searchTerm });

export const Unwrapped = Search; // only for testing purposes
export default connect(mapStateToProps)(Search);
