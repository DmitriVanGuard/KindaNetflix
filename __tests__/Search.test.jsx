import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Search from '../js/Search';

configure({ adapter: new Adapter() });

test(`Search renders correctly`, () => {
	const component = shallow(<Search />);
	expect(component).toMatchSnapshot();
});
