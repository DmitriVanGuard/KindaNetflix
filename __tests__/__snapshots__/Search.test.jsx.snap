// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="search"
>
  <header>
    <h1>
      KindaNetflix
    </h1>
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value="this is some sort of debug statement"
    />
  </header>
  <div />
</div>
`;
